### ISSUE TITLE
Repeated application of specific mutation fails on apply

### ISSUE THREAD

**Issue by trioderegion** [trioderegion/warpgate#102]
From @GamerFlix re: https://github.com/GamerFlix/effective-transferral/commit/c5095464ef87f94a9af6b802fa937f005605bf34

```js
{
    "embedded": {
        "ActiveEffect": {
            "Poisoned": {
                "label": "Poisoned",
                "icon": "icons/svg/biohazard.svg",
                "origin": "Item.IneExzr5ajNcnAI3",
                "duration": {
                    "rounds": 1,
                    "startTime": null,
                    "seconds": null,
                    "combat": null,
                    "turns": null,
                    "startRound": null,
                    "startTurn": null
                },
                "disabled": false,
                "changes": [],
                "tint": null,
                "transfer": false,
                "flags": {
                    "effective-transferral": {
                        "transferBlock": {
                            "button": false,
                            "chat": false,
                            "displayCard": false
                        },
                        "castData": {
                            "origin": "Scene.zV8tu6eWVcXTXomV.Token.j45lTkU7Cq9UpwiK.Item.vvqB7bByyV3eQ1qH"
                        },
                        "mutationKey": "mjlq78BqDsERnJld"
                    }
                }
            }
        }
    }
}
```

First time works, second time errors with: 
![image](https://user-images.githubusercontent.com/14878515/205373708-a3370bb1-716e-497a-87ad-a7da446de87f.png)

**Comment by GamerFlix** [comment]
Further context:

Issue was noticed as part of trying to workaround warpgate's handling of ids as comparison keys which lead to mutation's not reverted as was discovered here: https://github.com/GamerFlix/effective-transferral/issues/32

The workaround can be seen in the commit here where a randomId is saved in the flags and used as comparison key later. This works during the first application but not the second:
https://github.com/GamerFlix/effective-transferral/commit/c5095464ef87f94a9af6b802fa937f005605bf34

**Comment by trioderegion** [comment]
Related: #103 

103's failing was the prompt for using the flags as comparison keys, which appears to have presented another issue entirely.

**Comment by trioderegion** [comment]
Re-using the same `updates` object appears to be the root cause. This is unintended by design.  The following line of code fixes multiple mutations using the same updates object by first copying it to preserve its initial data.

`await warpgate.mutate(target, foundry.utils.duplicate(updates), {}, warpgateObject);`

I need to decide on a convention surrounding the ownership of the `updates` object -- should the user expect to be able to use it again to apply the same mutation to another token? Should the user be responsible for using a "clean" updates object for each mutation call? Should the update shimming/cleaning code create a duplicate of the updates object for internal use? Or some combination?

**Comment by krbz999** [comment]
> I need to decide on a convention surrounding the ownership of the `updates` object -- should the user expect to be able to use it again to apply the same mutation to another token? Should the user be responsible for using a "clean" updates object for each mutation call? Should the update shimming/cleaning code create a duplicate of the updates object for internal use? Or some combination?

My two cents: This can be answered rather simply by, "This is an unintended, undocumented, and breaking change." This is *a lot* of scripts that are currently broken for many different users all of a sudden, and can all be fixed by Warp Gate simply using `foundry.utils.duplicate`.

**Comment by GamerFlix** [comment]
> > I need to decide on a convention surrounding the ownership of the `updates` object -- should the user expect to be able to use it again to apply the same mutation to another token? Should the user be responsible for using a "clean" updates object for each mutation call? Should the update shimming/cleaning code create a duplicate of the updates object for internal use? Or some combination?
> 
> My two cents: This can be answered rather simply by, "This is an unintended, undocumented, and breaking change." This is *a lot* of scripts that are currently broken for many different users all of a sudden, and can all be fixed by Warp Gate simply using `foundry.utils.duplicate`.

If that explanation is to be used I would suggest fixing the "undocumented" part of the whole thing. I sure would have liked to see this in the patch notes :P

**Comment by krbz999** [comment]
Ideally it should not occur, to be blunt.

As for how users should be expected to use the `updates` object; exactly as before, which follows the similar use case of updating several documents with a single object (`Document#update`), which _can_ be looped.

**Comment by trioderegion** [comment]
Agreed. I'll try to get a fix in tomorrow morning. And clarify the ownership and lifetime of the various update objects that are interacted with in the documentation.

**Comment by GamerFlix** [comment]
While the object pased to mutate is no longer being modified the originally described issue still exists when the previously mentioned mutation key workaround is used. Same error as before. Apart from the object no longer being mutated nothing seems to have changed.
